keyboards/handwired/baredev/rev1/keymaps/manoshu/keymap.c

```diff
@@ -0,0 +1,131 @@
+/* Copyright 2021 Fernando "ManoShu" Rodrigues
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+#include "keymap_br_abnt2.h"
+#include "sendstring_br_abnt2.h"
+
+enum { 
+    LAYER_BASE, 
+    LAYER_FUNCTIONS 
+};
+
+#define L_FUNCT MO(LAYER_FUNCTIONS)
+#define W_CHOLD LGUI_T(KC_CAPS)
+#define SEND_WINDOWS(x) SEND_STRING(SS_LGUI(x))
+
+enum custom_keycodes {
+    KC_SNIP = SAFE_RANGE,  // Screen Snip (Shift + S)
+    KC_WRUN,               // Run (R)
+    KC_WLCK,               // Lock (L)
+    KC_WDSK,               // Desktop (D)
+    KC_WEXP,               // Explorer (E)
+    KC_WPSE,               // System Properties (Pause)
+    KC_WMGP,               // Magnify (+)
+    KC_WMGM,               // Magnify (-)
+    KC_WPRN,               // Project (P)
+    KC_WSPC,               // Input Select (Space)
+    KC_PSBR                // Ctrl-Pause/Break
+};
+
+layer_state_t layer_state_set_user(layer_state_t state) {
+    if (get_highest_layer(state) == LAYER_BASE) {
+        unregister_mods(MOD_MASK_GUI);
+    }
+    return state;
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    if(keycode == KC_WPRN || keycode == KC_WSPC) {
+        if (record->event.pressed) {
+
+            register_code(KC_LGUI);
+
+            if(keycode == KC_WPRN) {
+                tap_code(KC_P);
+            } else if (keycode == KC_WSPC) {
+                tap_code(KC_SPACE);
+            }
+        }
+    } else if (keycode == KC_PSBR) {
+        if (record->event.pressed) {
+            register_code(KC_LCTL);
+            tap_code(KC_PAUSE);
+            unregister_code(KC_LCTL);
+        }
```

Comment:
This can be replaced with `C(KC_PAUSE)` in the keymap itself.  Or as part of `tap_code16`

Reply:
Thank you. Somehow I missed those modifier functions. Made the changes to simplify those keycodes.
Code is now more easy to read too.